FirebaseStorage/Sources/FIRStorageDownloadTask.m

```diff
@@ -165,13 +165,15 @@ - (void)cancelWithError:(NSError *)error {
 - (void)pause {
   __weak FIRStorageDownloadTask *weakSelf = self;
   [self dispatchAsync:^() {
-    weakSelf.state = FIRStorageTaskStatePausing;
-    [weakSelf.fetcher stopFetching];
+    __strong FIRStorageDownloadTask *strongSelf = weakSelf;
+    strongSelf.state = FIRStorageTaskStatePausing;
+    [strongSelf.fetcher stopFetching];
     // Give the resume callback a chance to run (if scheduled)
-    [weakSelf.fetcher waitForCompletionWithTimeout:0.001];
-    weakSelf.state = FIRStorageTaskStatePaused;
-    FIRStorageTaskSnapshot *snapshot = weakSelf.snapshot;
-    [weakSelf fireHandlersForStatus:FIRStorageTaskStatusPause snapshot:snapshot];
+    dispatch_after(0.001 * NSEC_PER_SEC, strongSelf.dispatchQueue, ^{
```

Comment:
Would a simple call to dispatch a new block on the queue be enough here? The dispatch queue is configured to run tasks in order.

Reply:
Fixed this so it's not as silly. It turns out the `pause` method needs to wait for GTMSessionFetcher to finish its last update, or the internal state of the download task will be incorrect.